api/client/client.go

```diff
@@ -27,6 +29,27 @@ type Client struct {
 	conn       *grpc.ClientConn
 }
 
+type ctxKey string
+
+const AuthKey = ctxKey("ffstoken")
+
+type TokenAuth struct {
+	secure bool
+}
+
+func (t TokenAuth) GetRequestMetadata(ctx context.Context, _ ...string) (map[string]string, error) {
+	md := map[string]string{}
+	token, ok := ctx.Value(ctxKey(AuthKey)).(string)
+	if ok && token != "" {
+		md["X-ffs-Token"] = token
+	}
+	return md, nil
+}
+
+func (t TokenAuth) RequireTransportSecurity() bool {
+	return t.secure
+}
+
```

Comment:
Moved the auth stuff to the client so it can be used anywhere the client is used